src/hammer-vlsi/hammer_vlsi/hammer_tool.py

```diff
@@ -914,6 +914,49 @@ def get_bumps(self) -> Optional[BumpsDefinition]:
             pitch=self.get_setting("vlsi.inputs.bumps.pitch"),
             cell=self.get_setting("vlsi.inputs.bumps.cell"), assignments=assignments)
 
+    def get_pin_assignments(self) -> List[PinAssignment]:
+        """
+        Get a list of pin assignments in accordance with settings in the Hammer IR.
+        :return: A potentially empty list of PinAssigments.
+        """
+        pin_mode = str(self.get_setting("vlsi.inputs.pin_mode"))  # type: str
+        if pin_mode == "none":
+            return []
+        elif pin_mode != "generated":
+            self.logger.error(
+                "Invalid pin_mode {mode}. Using none pin mode.".format(mode=pin_mode))
+            return []
+
+        # Generated pin mode needs to ingest the assignments
+        assigns = []  # type: List[PinAssignment]
+        for raw_assign in self.get_setting("vlsi.inputs.pin.assignments"):
+            pins = str(raw_assign["pins"])  # type: str
+            side = None if not "side" in raw_assign else raw_assign["side"]
+            if not (side == None or side == "top" or side == "bottom" or side == "right" or side == "left") :
+                self.logger.warning("Pins {p} have invalid side {s}. Assuming pins will be handled by CAD tool.".format(p=pins, s=side))
+                continue
+            macro = False if not "macro" in raw_assign else raw_assign["macro"]
+            layers = [] if not "layers" in raw_assign else raw_assign["layers"]
+            if macro:
+                if len(layers) != 0 or side != None:
+                    self.logger.warning("Pins {p} assigned as a macro pin with layers or side. Assuming pins are macro pins and ignoring layers and side.".format(p=pins))
+                    assigns.append(PinAssignment(pins=pins, side=None, layers=[], macro=macro))
+                    continue
+            else:
+                if len(layers) == 0 or side == None:
+                    self.logger.warning("Pins {p} assigned without layers or side. Assuming pins will be handled by CAD tool.".format(p=pins))
+                    # No pin appended
+                    continue
+            stackup = self.get_stackup()
+            for layer in layers:
+                direction = stackup.get_metal(layer).direction
+                if not((direction == RoutingDirection.Horizontal and (side == "left" or side == "right")) or
```

Comment:
style

Reply:
This is a complex boolean expression. I wrote it out verbosely so that it is easy to understand the intent.
What is the style convention I should follow?

Reply:
Maybe break them out like

```python
is_horizontal = direction == RoutingDirection.Horizontal and side == "left" or side == "right"
is_vertical = direction == RoutingDirection.Vertical and side == "top" or side == "bottom"
is_redis = direction == RoutingDirection.Redistribution
if not (is_horizontal or is_vertical or is_redis):
  ...
```